enumerable.rb

```diff
@@ -0,0 +1,202 @@
+    elsif !block_given? && arg.nil?
+      counter = my_each.length
+    else
+      counter = my_select { |i| i == arg }.length
+    end
+    counter
+  end
+
+  ## my_map method
+  def my_map(factor = nil)
+    return to_enum(:my_map) unless block_given? || !factor.nil?
+
+    new_arr = []
+    if factor.nil?
+      my_each { |i| new_arr << yield(i) }
+    else
+      my_each { |i| new_arr << factor.call(i) }
+    end
+    new_arr
+  end
+
+  def my_inject(value_one = nil, value_two = nil)
+    if (!value_one.nil? && value_two.nil?) && (value_one.is_a?(Symbol) || value_one.is_a?(String))
+      value_two = value_one
+      value_one = nil
+    end
+    if !block_given? && !value_two.nil?
+      my_each { |i| value_one = value_one.nil? ? i : value_one.send(value_two, i) }
+    else
+      my_each { |i| value_one = value_one.nil? ? i : yield(value_one, i) }
+    end
+    value_one
+  end
+  # rubocop:enable Metrics/CyclomaticComplexity
+  # rubocop:enable Metrics/PerceivedComplexity
+  # rubocop:enable Lint/ToEnumArguments
+end
+
+## multiply_els method
+def multiply_els(arr)
+  arr.my_inject(1, '*')
+end
+
+# # my_each method
+# puts '---- my_each ----'
+# puts([2, 5, 6, 7].my_each { |x| x })
+# puts([2, 5, 6, 7, nil].my_each { |x| x })
+# puts([2, 5, 6, 7, nil, 'hello'].my_each { |x| x })
+# puts((0..10).my_each { |x| x })
+# puts({ 'name' => 'John', 'age' => '21', 'adress' => 'USA' }.my_each { |x| x })
+
+# # my_each_index
+# puts '---- my_each_index ----'
+# [2, 5, 6, 7].my_each_with_index { |x, i| puts "#{i} : #{x}" }
+# %w[Victor Igor Microverse Program].my_each_with_index { |x, i| puts x if (i % 2).zero? }
+
+# # my_select
+# puts '---- my_select ----'
+# puts([2, 5, 6, 7].my_select { |n| n })
+
+# # my_all
+# puts '---- my_all ----'
+# string_arr = ['world', 'hello', 'help']
+# puts string_arr.my_all? { |word| word.length <= 3 }
+# puts string_arr.my_all? { |word| word.length >= 4 }
+# puts string_arr.my_all?(/l/)
+# puts [1, 2, 3.14].my_all?(Numeric)
+# puts [1, 'the', 3.14].my_all?(Numeric)
+# puts [].my_all?
+# puts [1, 2, 3].my_all?
+
+# # # my_any
+# puts '---- my_any ----'
+# string_arr = ['world', 'hello', 'help']
+# puts string_arr.my_all? { |word| word.length <= 3 }
+# puts string_arr.my_all? { |word| word.length >= 3 }
+# puts string_arr.my_all?(/h/)
+# puts [1, 2, 3.14].my_all?(Numeric)
+# puts [1, 'the', 3.14].my_all?(Numeric)
+# puts [].my_all?
+# puts [1, 2, 3].my_all?
+
+# # my_none
+# puts '---- my_none ----'
+# string_arr = ['world', 'hello', 'help']
+# puts string_arr.my_all? { |word| word.length == 3 }
+# puts string_arr.my_all? { |word| word.length >= 4 }
+# puts string_arr.my_all?(/o/)
+# puts [1, 2, 3.14].my_all?(Integer)
+# puts [1, 'the', 3.14].my_all?(Numeric)
+# puts [].my_all?
+# puts [1, 2, 3].my_all?
+
+# # my_count method
+# puts '---- my_count ----'
+# puts((1..3).my_count(3){|n| n})
+
+# # my_map method
+# puts '---- my_map ----'
+# puts([2, 5, 7, 4, 2].my_map { |x| x<3 })
+
```

Comment:
- [ ] Please remove these commented codes as well as other commented codes in your project.